model/project_configs.go

```diff
@@ -33,6 +32,22 @@ type ProjectConfig struct {
 	PeriodicBuilds         []PeriodicBuildDefinition      `yaml:"periodic_builds,omitempty" bson:"periodic_builds,omitempty"`
 }
 
+type HeadlessProjectConfig struct {
```

Comment:
Is the reason we need this because ID is a duplicated field across ProjectConfig and ParserProject or something and so it doesn't work to have ProjectConfig within the struct used for validating strictly?

I'm pretty against having this struct. It's a lot of code duplication for something that's only used in one place, and "Headless" isn't very informative (and is a bit morbid haha).  Alternatives could be:

* Have a ProjectConfig structure called ProjectConfigFields (or something similar); have this called inline for project config. Add comments about why we're doing this. Instead of the GetHeadless / GetProject functions, could just return ProjectConfig.ProjectConfigFields or whatever.
* Just add each ProjectConfig field that we need to validate into the unmarshalWithStrict field directly. Maybe add a comment to ProjectConfig mentioning to add these there. (Similar amount of code duplication but now it's restricted to the code that needs it.)

Reply:
I had initially gone with your first solution but backed out due to the amount of refactoring in other files / tests that was required, but I get where you're coming from. I've added it back and it's actually cleaner than I expected. Also removed the "Headless" naming in favor of 'ProjectConfigFields'. Honestly that was just me being a little cheeky and deep down I knew that name wouldn't fly haha.

Reply:
> Sorry to balloon this but I do think there are a few more places to audit to make sure IsVersionControlEnabled is used in the right places. If you'd like we could split that to a different ticket.

I've audited the usages of CreateProjectConfig and there shouldn't be any instances of it being called if a project ref has version control disabled at this point.